From 5739c7ff255c5bff66712db693163ac54ebf5405 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:47 -0500
Subject: [drm] radeon: properly document reloc priority mask

Message-id: <1418852380-13061-37-git-send-email-rclark@redhat.com>
Patchwork-id: 102168
O-Subject: [RHEL7 drm 36/89] drm/radeon: properly document reloc priority mask
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.16.4

commit d7df33d81dfede0b5d6fde96cdda40cd17e32921

Author:     Christian Koenig <christian.koenig@amd.com>
AuthorDate: Fri Aug 15 11:52:53 2014 +0200
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:19:45 2014 -0500

    drm/radeon: properly document reloc priority mask

    commit 701e1e789142042144c8cc10b8f6d1554e960144 upstream.

    Instead of hard coding the value properly document
    that this is an userspace interface.

    No intended functional change.

    Signed-off-by: Christian Koenig <christian.koenig@amd.com>
    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/radeon/radeon_cs.c b/drivers/gpu/drm/radeon/radeon_cs.c
index ae763f6..8f7d56f 100644
--- a/drivers/gpu/drm/radeon/radeon_cs.c
+++ b/drivers/gpu/drm/radeon/radeon_cs.c
@@ -132,7 +132,8 @@ static int radeon_cs_parser_relocs(struct radeon_cs_parser *p)
    * the buffers used for read only, which doubles the range
    * to 0 to 31. 32 is reserved for the kernel driver.
    */
-  priority = (r->flags & 0xf) * 2 + !!r->write_domain;
+  priority = (r->flags & RADEON_RELOC_PRIO_MASK) * 2
+      + !!r->write_domain;
 
   /* the first reloc of an UVD job is the msg and that must be in
      VRAM, also but everything into VRAM on AGP cards to avoid
diff --git a/include/uapi/drm/radeon_drm.h b/include/uapi/drm/radeon_drm.h
index 1cc0b61..79719f9 100644
--- a/include/uapi/drm/radeon_drm.h
+++ b/include/uapi/drm/radeon_drm.h
@@ -942,6 +942,7 @@ struct drm_radeon_cs_chunk {
 };
 
 /* drm_radeon_cs_reloc.flags */
+#define RADEON_RELOC_PRIO_MASK  (0xf << 0)
 
 struct drm_radeon_cs_reloc {
  uint32_t  handle;
-- 
1.7.1